### PR TITLE
Create Ruff reusable action

### PR DESCRIPTION
**Pull request by forrestfwilliams**
[Ruff](https://github.com/astral-sh/ruff) is a crazy fast linting and formatting tool that we think is worth using to perform static analysis checks on our repositories. This PR adds a Ruff reusable action that will check three things:

1. General Ruff linting rules
2. isort-based import order
3. `ruff format` style formating

For posterity, most errors detected by this action can be fixed by running the following three commands in a repository's home directory:

```bash
ruff check --fix .
ruff check --select I --fix .
ruff format .
```

For the action to conform to the Tools team's style preferences, the following lines will need to be add to a project's `pyproject.toml` file:

```toml
[tool.ruff]
line-length = 120
src = ["src", "tests"]

[tool.ruff.format]
indent-style = "space"
quote-style = "single"

[tool.ruff.lint.isort]
case-sensitive = true
lines-after-imports = 2
```

### CODE REVIEW

**Review by jtherrmann — COMMENTED** [review]
LGTM. We'll also want to add a usage example to the README. We may also want to test it on the `hyp3-isce2` repo before merging this PR (I think the syntax is `@ruff` for specifying that the action is on the `ruff` branch).

**Comment by forrestfwilliams** [comment]
@jtherrmann updated the readme and reformatted the workflow file

File: README.md
Comment (jhkennedy) [inline README.md:239]: Is it worth adding the corrective action to the README as well?

> For posterity, most errors detected by this action can be fixed by running the following three commands in a repository's home directory:
  ```shell
  ruff check --fix .
  ruff check --select I --fix .
  ruff format .
  ```

**Review by jhkennedy — APPROVED** [review]
Looks good! One minor comment I'm happy to defer to you on.

**Review by jhkennedy — APPROVED** [review]

**Pull request merged**
